File: lc_soln/0853_car_fleet/solution.py
from typing import List


class Solution:
    def carFleet(self, target: int, position: List[int], speed: List[int]) -> int:
        pos_and_spd = sorted(zip(position, speed), key=lambda x: -x[0])
        stack = []
        for pos, spd in pos_and_spd:
            time = (target - pos) / spd
            if stack and time <= stack[-1]:
                continue
            stack.append(time)
        return len(stack)


def main(): ...


if __name__ == "__main__":
    main()

File: lc_soln/0853_car_fleet/solution.cs
public class Solution
{
    public int CarFleet(int target, int[] position, int[] speed)
    {
        (int, int)[] posAndSpd = position.Zip(speed, (pos, spd) => (pos, spd)).ToArray();
        Array.Sort(
            posAndSpd,
            ((int pos, int spd) a, (int pos, int spd) b) => b.pos.CompareTo(a.pos)
        );
        var stack = new Stack<double>();
        foreach ((int pos, int spd) in posAndSpd)
        {
            double time = (double)(target - pos) / (double)spd;
            if (stack.Count > 0 && time <= stack.Peek())
            {
                continue;
            }
            stack.Push(time);
        }
        return stack.Count;
    }
}
